skimage/graph/tests/test_mcp.py

```diff
@@ -186,3 +186,28 @@ def _test_random(shape):
     for end in ends:
         m.traceback(end)
     return a, costs, offsets
+
+
+def test_max_costs():
+    image = np.array([[1, 1, 2, 2], [2, 1, 1, 3], [3, 2, 1, 2], [2, 2, 2, 1]])
+    with expected_warnings(['Upgrading NumPy' + warning_optional]):
+        m = mcp.MCP(image, fully_connected=True)
+
+    destinations = [[0, 0], [3, 3]]
+
+    costs, traceback = m.find_costs(destinations)
+
+    assert_array_equal(costs, [[1, 2, 4, 5], [3, 2, 3, 5], [5, 4, 2, 3], [6, 4, 3, 1]])
+    assert_array_equal(
+        traceback, [[-1, 4, 2, 2], [6, 7, 4, 2], [5, 6, 0, 1], [3, 5, 3, -1]]
+    )
+
+    costs_limited, traceback_limited = m.find_costs(destinations, max_cost=2)
+
+    assert_array_equal(
+        costs_limited,
+        [[1, 2, 4, 5], [3, 2, 3, np.inf], [np.inf, 4, 2, 3], [6, 4, 3, 1]],
```

Comment:
This looks valid but it might be better to chose a example where the path-finding is forced to go around the barrier created by `max_cost`? For this case, `max_cost` only affects the location of the barrier itself and doesn't assert that the algorithm can't go through the barrier. Does this explanation make sense?

Reply:
Agree that the example should also show that the path-finding goes around the barrier. I modified the example.